Quadriga/src/main/java/edu/asu/spring/quadriga/web/config/SimpleSignInAdapter.java

```diff
@@ -0,0 +1,127 @@
+import edu.asu.spring.quadriga.web.login.QuadrigaGrantedAuthority;
+import edu.asu.spring.quadriga.web.manageusers.beans.AccountRequest;
+
+/**
+ * 
+ * This class authorizes (assigns appropriate roles) users authenticated by
+ * Github.
+ * 
+ * @author Chiraag Subramanian
+ *
+ */
+
+public final class SimpleSignInAdapter implements SignInAdapter {
+
+    private final Logger logger = LoggerFactory.getLogger(SignInAdapter.class);
+
+    private IUserManager userManager;
+    private IUserHelper userHelper;
+
+    public SimpleSignInAdapter(IUserManager userManager, IUserHelper userHelper) {
+        this.userManager = userManager;
+        this.userHelper = userHelper;
+    }
+
+    /**
+     * This method checks if Quadriga contains details of the user authenticated
+     * by Github. If the user details are present (which indicates the user is
+     * an authorized user), appropriate roles are assigned, Else if the user
+     * details are not present (which indicates the user is an unregistered /
+     * not approved user, restricted access is assigned)
+     */
+
+    @Override
+    public String signIn(String userId, Connection<?> connection, NativeWebRequest request) {
+        List<GrantedAuthority> authorities = new ArrayList<>();
+        SocialSignInStatus messageType = null;
+        IUser user = (IUser) userManager.findUserByProviderUserId(connection.getKey().getProviderUserId(),
+                connection.getKey().getProviderId());
+        // if user details is not present in the database, create a database
+        // record for the user.
+        if (user == null) {
+            user = userHelper.createUser(connection);
+            try {
+                
+                if(userManager.checkUserRequestExists(user.getUserName())){
+                    messageType = SocialSignInStatus.REGISTRATION_SUCCESS;
+                }
+                else{
+                    AccountRequest accountRequest = new AccountRequest();
+                    accountRequest.setUsername(user.getUserName());
+                    accountRequest.setName(user.getName());
+                    accountRequest.setEmail(user.getEmail());
+                    accountRequest.setSocialSignIn(true);
+                    accountRequest.setProvider(user.getProvider());
+                    accountRequest.setUserIdOfProvider(user.getUserIdOfProvider());
+            
+                    if(userManager.addNewUser(accountRequest) ){
+                        messageType = SocialSignInStatus.REGISTRATION_SUCCESS;
+                    }
+                }
+            } catch (QuadrigaStorageException | UsernameExistsException e) {
+                logger.error("Could not add user.", e);
+                messageType = SocialSignInStatus.REGISTRATION_FAILED;
+            } catch (QuadrigaNotificationException e) {
+                logger.error("Could not notify admin about the new user.", e);
+            } 
```

Comment:
catch them where they are thrown. if they are thrown more than once and the two statements that throw them are a couple of lines apart, you have to catch them twice in most cases.